compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -0,0 +1,77 @@
+package nodeinventorizer
+
+import (
+	timestamp "github.com/gogo/protobuf/types"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/scanner/database"
+	scannerV1 "github.com/stackrox/scanner/generated/scanner/api/v1"
+	"github.com/stackrox/scanner/pkg/analyzer/nodes"
+)
+
+// NodeInventorizer is the interface that defines the interface a scanner must implement
+type NodeInventorizer interface {
+	Scan(nodeName string) (*storage.NodeInventory, error)
+}
+
+// NodeInventoryCollector is the implementation of NodeInventorizer
+type NodeInventoryCollector struct {
+}
+
+// Scan scans the current node and returns the results as storage.NodeInventory object
+func (n *NodeInventoryCollector) Scan(nodeName string) (*storage.NodeInventory, error) {
+	componentsHost, err := nodes.Analyze(nodeName, "/host/", false)
+	log.Info("Finished node inventory /host scan")
+	if err != nil {
+		log.Errorf("Error scanning node /host inventory: %v", err)
+		return nil, err
+	}
+	log.Infof("Components found under /host: %v", componentsHost)
+
+	protoComponents := protoComponentsFromScanComponents(componentsHost)
+
+	m := &storage.NodeInventory{
+		NodeName:   nodeName,
+		ScanTime:   timestamp.TimestampNow(),
+		Components: protoComponents,
+	}
+	return m, nil
+}
+
+func protoComponentsFromScanComponents(c *nodes.Components) *scannerV1.Components {
```

Comment:
I believe we discussed probably moving this inside the Scanner repo, right?

Reply:
Yes, we did. I have created ROX-14029 to track this change for the future. Thank you for reminding me 👍